hive-metastore/src/main/java/org/apache/iceberg/hive/ClientPool.java

```diff
@@ -114,8 +114,9 @@ private C get() throws InterruptedException {
           if (!clients.isEmpty()) {
             return clients.removeFirst();
           } else if (currentSize < poolSize) {
+            C client = newClient();
             currentSize += 1;
```

Comment:
This is real issue here right? We have the possibility of incrementing current size even if "newClient" failed in the old code? If so this seems like the right fix to me

Reply:
Yeah, that's right.